openlibrary/core/bookshelves.py

```diff
@@ -346,14 +421,28 @@ def get_filtered_reading_log_books(
                 facet=False,
                 # Putting these in fq allows them to avoid user-query processing, which
                 # can be (surprisingly) slow if we have ~20k OR clauses.
-                extra_params=[('fq', f'key:({" OR ".join(reading_log_work_keys)})')],
+                extra_params=[('fq', filter_query)],
             )
             total_results = solr_resp.num_found
+            solr_docs = solr_resp.docs
+
+            if show_editions:
+                edition_data = get_solr().get_many(
+                    [f'{key}' for key in work_to_edition_keys.values()],
```

Comment:
```suggestion
                    work_to_edition_keys.values(),
```

Reply:
Change implemented, with an addendum. Instead of fetching all editions in the reading log now, we're only fetching the ones that match up to works.